ui/components/Dataset/index.js

```diff
@@ -188,45 +189,45 @@ export default function Dataset({
   const [queue, setQueue] = useState(null);
   const { count = 0, results = [] } = data;
   const filtersSelected = Object.keys(getSelections).length > 0;
+  const router = useRouter();
 
-  function pushToQueue() {
+  useEffect(() => {
     setQueue(query);
-  }
+  }, [query]);
+  useEffect(() => {
+    async function getData() {
+      if (loading || !queue) {
+        return;
+      }
 
-  async function getData() {
-    if (loading || !queue) {
-      return;
-    }
+      const DEFAULT_SORT = {
+        score: 'desc',
+        metadata_modified: 'desc',
+      };
 
-    const DEFAULT_SORT = {
-      score: 'desc',
-      metadata_modified: 'desc',
-    };
+      const { q, page, sort = DEFAULT_SORT, ...filters } = queue;
+      const params = {
+        q,
+        page,
+        sort,
+        filters,
+      };
 
-    const { q, page, sort = DEFAULT_SORT, ...filters } = queue;
-    const params = {
-      q,
-      page,
-      sort,
-      filters,
-    };
+      try {
+        setLoading(true);
+        setQueue(null);
+        const res = await axios.post('/api/refresh-list', params);
+        setData(res.data);
+      } catch (err) {
+        console.error(err);
 
-    try {
-      setLoading(true);
-      setQueue(null);
-      const res = await axios.post('/api/refresh-list', params);
-      setData(res.data);
-    } catch (err) {
-      console.error(err);
-      const router = useRouter();
-      router.reload(window.location.pathname);
-    } finally {
-      setLoading(false);
+        router.reload(window.location.pathname);
+      } finally {
+        setLoading(false);
+      }
     }
-  }
-
-  useEffect(() => pushToQueue(), [query]);
-  useEffect(() => getData(), [queue, loading]);
+    getData();
```

Comment:
Is there a particular reason for the define a function then immediately call it thing going on here, as opposed to just running the code?

Reply:
Fair play, will do!

Reply:
Oh no my bad (it's been a while since I looked at this), it's because we're using an async function inside the hook.
```
· · ·
// ❌ don't do this
useEffect(async () => {
  const data = await fetchData();
}, [fetchData])
· · ·
```

The issue here is that the first argument of `useEffect` is supposed to be a function that returns either nothing (undefined) or a function (to clean up side effects). But an async function returns a Promise, which can't be called as a function.

Reply:
Fair, I suspected that might be something to do with it.